kentikapi/internal/resources/alerting.go

```diff
@@ -15,19 +16,21 @@ type AlertingAPI struct {
 }
 
 // NewAlertingAPI is constructor.
-func NewAlertingAPI(transport api_connection.Transport) *AlertingAPI {
+func NewAlertingAPI(transport api_connection.Transport, logPayloads bool) *AlertingAPI {
 	return &AlertingAPI{
-		BaseAPI{Transport: transport},
+		BaseAPI{Transport: transport, LogPayloads: logPayloads},
 	}
 }
 
 func (a *AlertingAPI) CreateManualMitigation(ctx context.Context, mm models.ManualMitigation) error {
 	payload := api_payloads.ManualMitigationToPayload(mm)
+	utils.LogPayload(a.LogPayloads, "CreateManualMitigation Kentik API request", payload)
```

Comment:
Format of logging string doesn't need to be the same for TF providers and Go SDK. However, format of payload logs for API v5 and API v6 methods should be similar (keyword: consistency).

In #85 we have following format
> Kentik API request - /kentik.synthetics.v202101beta1.SyntheticsAdminService/CreateTest - test:{name:"example-> test-1" [...] }

So I suggest to make following format for API v5:
`"Kentik API request - <method> <URL> - <payload>"`
`"Kentik API request - POST localhost:1234/api/v5/users - {"user": {"name": "foo"}}"` 

All that information is available at BaseAPI level, so I think logging should be put there to reduce amount of code. The same with response logging.

Reply:
I very much agree. If there are additional information that needs to be logged at higher levels, multiple messages may be logged for a single requests.